### PR TITLE
feat(workspaces): Integrating BE with FE

### PR DESCRIPTION
**Pull request by foochifa**
## Problem

Integrating the frontend with all the workspaces BE works that @hanstirtaputra have done

Context can be found in #4222 

## Solution

Added workspace service calls and useQueries.

Integrated workspace components and properly use `WorkspaceContext` from workspace searching and filtering done by @justynoh #5458 

Also fixed BE call for form delete to also delete from workspaces

**Breaking Changes** 

- [ ] No - this PR is backwards compatible  to workspaces branch only

## Before & After Screenshots

**BEFORE**:

**AFTER**:

### CODE REVIEW

**Comment by foochifa** [comment]
Will fix test cases but welcome for review first

Also Note: Pushing this to a feature branch and not develop yet

**Comment by foochifa** [comment]
@kenjin-work https://www.figma.com/file/CMq0CJtawn7NGM7Zdpi8Ti/Form-Design-Master?type=design&node-id=185-118&mode=design&t=ldzjPrkjHCZRbSMk-0

**Comment by kenjin-work** [comment]
UI review first:

There's a small gap on the right that I don't see in the Figma:
image

No tick on the current workspace that the form is in (in this case this form's belongs to `test` but it's not showing)
image

Backend (can be addressed in another PR):
Workspaces with duplicate names can be created. When admin selects workspace to move to this can lead to confusion.

File: frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/RowActionsDropdown.tsx [hunk 1]
```diff
@@ -1,26 +1,73 @@
...
+  formMeta,
+}: {
+  setIsMoveWorkspace: Dispatch<SetStateAction<boolean>>
+  formMeta: AdminDashboardFormMetaDto
+}) => {
+  const { handleMoveForm } = useRowAction(formMeta)
+  const { workspaces } = useWorkspaceContext()
+
+  const handleWorkspaceClick = (destWorkspace: Workspace) =>
+    handleMoveForm(destWorkspace._id.toString(), destWorkspace.title)
```
Comment (kenjin-work) [inline frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/RowActionsDropdown.tsx]: Should use a `useCallback` here (might be more efficient).

**Review by kenjin-work — COMMENTED** [review]
One smol comment for code. Mostly LGTM.

File: frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/RowActionsDropdown.tsx (hunk 1 above)
Comment (foochifa) [reply frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/RowActionsDropdown.tsx]: Thanks @kenjin-work! [ad5b50e](https://github.com/opengovsg/FormSG/pull/6560/commits/ad5b50e7937face50761c8b0d1f84f68d3ac0c4c)

**Comment by foochifa** [comment]
Heya @KenLSM @LinHuiqing 

Kenjin reviewed this before he left, and have addressed his comments. 

However, i have made certain changes since and would like another pair of eyes!

1. To show current workspace for form, have made a util method within the context to find the workspace the form belongs in. Will then populate `RowActionDrawer` or `RowActionDropdown`
2. Implemented `RowActionDrawer` for mobile view and fixed some grid issues to ensure mobile is okay

Let me know if you need more context (Will also update the manual test section)! After this, there's only 1 more issue left which is to delete form from workspace when removed as collaborator!.

**Review by KenLSM — APPROVED** [review]
LGTM!

**Pull request merged**